module/VuFind/src/VuFind/Db/Table/User.php

```diff
@@ -104,12 +104,35 @@ public function createRowForUsername($username)
         return $row;
     }
 
+    /**
+     * Retrieve a user object from the database based on the given field name.
+     *
+     * @param string $fieldName  Field name
+     * @param string $fieldValue Field value
+     *
+     * @return ?UserRow
+     */
+    public function getByField($fieldName, $fieldValue)
+    {
+        switch ($fieldName) {
+            case 'id':
+                return $this->getById($fieldValue);
+            case 'username':
+                return $this->getByUsername($fieldValue, false);
+            case 'cat_id':
+                return $this->getByCatalogId($fieldValue);
+            case 'email':
```

Comment:
It is not needed but I left it since it might be used in the future. However, I can delete it if you prefer that.

Reply:
I'd leave it out since there's no guarantee at all for it to be unique.

Reply:
Being unique is not necessary for other possible use cases for this method. But off course if such a use case will occur it can be added again. So I will remove it for now.

Reply:
Done